xrpl4j-integration-tests/src/test/java/org/xrpl/xrpl4j/tests/AbstractIT.java

```diff
@@ -75,19 +76,33 @@ public abstract class AbstractIT {
 
   public static final String SUCCESS_STATUS = TransactionResultCodes.TES_SUCCESS;
 
-  protected static XrplEnvironment xrplEnvironment = XrplEnvironment.getConfiguredEnvironment();
-
   protected final Logger logger = LoggerFactory.getLogger(this.getClass());
-
-  protected final XrplClient xrplClient = xrplEnvironment.getXrplClient();
+  protected final XrplClient xrplClient = xrplClient();
   protected final WalletFactory walletFactory = DefaultWalletFactory.getInstance();
 
+  protected static XrplEnvironment xrplEnvironment;
+
+  /**
+   * Helper method to initialize the {@link XrplEnvironment} and to also allow sub-classes to use a different
+   * environment. This construction allows sub-classes to choose an environment that doesn't spin-up a test container.
+   *
+   * @return An {@link XrplEnvironment}.
+   */
+  @BeforeAll
```

Comment:
@mukulljangid @nkramer44 This is the pattern I had in mind. It allows three important things for our ITs:

1. It allows this `AbstractIT` or any sub-classes to use the `xrplEnvironment` variable directly so that if a developer forgets to call `xrplEnvironment()` as a function, things will still work.

2. It allows a potential future where we might run these ITs in parallel.

3. It enables the goal that @nkramer44 was hoping for which is that any IT that doesn't need a Testcontainer-based rippled won't spin one up un-necessarily. You can see now that `NfTokenIT` doesn't use a test-container (due to the override of `initXrplEnvironment()` in place in that class) when executed on its own. However, when all tests are run together, the _other_ tests will initialize the xrplEnvironment correctly using the variant in `AbstractIT`.